api/common/common_types.go

```diff
@@ -0,0 +1,120 @@
+	// to the O2IMS DNS FQDN (e.g., o2ims.apps.<cluster domain name>).  This is to ensure that the certificate can be
+	// used as the ingress certificate as well as the outgoing client certificate.
+	//+optional
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="TLS certificate"
+	SecretName *string `json:"secretName"`
+}
+
+// +kubebuilder:object:generate=true
+// AuthType defines the authorization type used for authentication.
+type AuthType string
+
+const (
+	// ServiceAccount represents authentication using a Kubernetes ServiceAccount token. This method leverages
+	// a Kubernetes ServiceAccount to authenticate requests, typically used for internal cluster communication
+	// where the client presents a JWT token issued by the Kubernetes API server.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="ServiceAccount Authentication"
+	ServiceAccount AuthType = "ServiceAccount"
+
+	// Basic represents authentication using HTTP Basic Authentication. This method uses a username and password
+	// combination to authenticate requests, typically encoded in the Authorization header of HTTP requests.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Basic Authentication"
+	Basic AuthType = "Basic"
+
+	// OAuth represents authentication using an OAuth2-based mechanism. This method involves acquiring a token
+	// from an OAuth2 server, which is then used to authenticate requests, typically via a Bearer token in the
+	// Authorization header. Configuration details for OAuth are specified in the OAuthConfig struct.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OAuth Authentication"
+	OAuth AuthType = "OAuth"
+)
+
+// +kubebuilder:object:generate=true
+// AuthConfig defines the configuration for different authentication types.
+// This struct encapsulates the settings required for ServiceAccount, Basic, and OAuth authentication mechanisms.
+type AuthConfig struct {
+	// Type specifies the authentication type to be used (e.g., ServiceAccount, Basic, or OAuth).
+	//+kubebuilder:validation:Enum=ServiceAccount;Basic;OAuth
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Authentication Type",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:select:ServiceAccount","urn:alm:descriptor:com.tectonic.ui:select:Basic","urn:alm:descriptor:com.tectonic.ui:select:OAuth"}
+	Type AuthType `json:"type"`
+
+	// ServiceAccountName specifies the name of the Kubernetes ServiceAccount to use for authentication.
+	// This field is required when Type is set to "ServiceAccount". The ServiceAccount must exist in the
+	// current namespace, and its associated token will be used for authentication.
+	//+optional
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Service Account Name",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+	ServiceAccountName *string `json:"serviceAccountName,omitempty"`
```

Comment:
I would leave this out.   The hardware-manager plugin should be instantiated with a service account when the operator creates it (e.g., like we do for the alarm-server, resource-server, etc... they each have there own service account).